src/core/ColorPicker/ColorInputPanel.tsx

```diff
@@ -63,15 +64,35 @@ export const ColorInputPanel = React.forwardRef(
     React.useEffect(() => {
       if (currentFormat === 'hsl') {
         const hsl = activeColor.toHslColor();
+        const a =
+          hsl.a?.toFixed(2) ?? (activeColor.getAlpha() / 255).toFixed(2);
         // use activeHue to preserve hue for 0,0,0 edge case
-        setInput([hsvColor.h.toString(), hsl.s.toString(), hsl.l.toString()]);
+        setInput([
+          hsvColor.h.toString(),
+          hsl.s.toString(),
+          hsl.l.toString(),
+          a.toString(),
+        ]);
       } else if (currentFormat === 'rgb') {
         const rgb = activeColor.toRgbColor();
-        setInput([rgb.r.toString(), rgb.g.toString(), rgb.b.toString()]);
-      } else {
-        setInput([activeColor.toHexString()]);
+        const a =
+          rgb.a?.toFixed(2) ?? (activeColor.getAlpha() / 255).toFixed(2);
+        setInput([
+          rgb.r.toString(),
+          rgb.g.toString(),
+          rgb.b.toString(),
+          a.toString(),
+        ]);
       }
-    }, [activeColor, hsvColor.h, currentFormat]);
+    }, [activeColor, hsvColor.h, currentFormat, showAlpha]);
+
+    const [hexInput, setHexInput] = React.useState(
```

Comment:
Where did this come from?

Your commit says it's to fix the controlled vs uncontrolled warning, but it's not working (I am still seeing the warning). That warning was appearing because you are initializing your inputs with undefined (`[]` instead of `['', '', '']`).

One thing you can also do is just add `?? ''` when passing the `value` to the inputs

Reply:
Cleaned up ColorInputPanel with these changes